blockchains/erc20/lib/contract_overwrite.js

```diff
@@ -23,87 +19,67 @@ class ContractOverwrite {
-    return contractOverwrite.oldAddresses.includes(event.contract);
-  }
-
-  editAddressAndAmount(event, contractOverwrite) {
-    const multiplier = contractOverwrite.mapAddressToMultiplier[event.contract];
-    event.contract = contractOverwrite.newContract;
-    /**
-     * Note 1: Whether we should round here is up for discussion. The amounts in our pipeline are 'float' anyways but up until this feature
-     * actual values have always been Integers. Choosing to round for simplicity and backwards compatibility.
-     *
-     * Note 2: The decision whether to divide one of the contract values or multiply the other is also up for discussion. By dividing (and rounding)
-     * we loose precision and possibly mis-represent small amounts on the affected contract. The other possible decision is to multiply
-     * amounts on the contract using smaller amounts but in this way we may generate too huge values. We are choosing the first option.
-     */
-    event.value = Math.floor(event.value * multiplier);
-    const bigNumber = new BigNumber(event.valueExactBase36, 36).times(multiplier).integerValue();
-    event.valueExactBase36 = bigNumber.toString(36);
-  }
-
-  async getPastEventsExactContracts(web3, fromBlock, toBlock, timestampsCache) {
-    let resultsAggregation = [];
-
-    for (const contractOverwrite of this.contractsOverwriteArray) {
-      const events = await getPastEvents(web3, fromBlock, toBlock, contractOverwrite.oldAddresses, timestampsCache);
-      resultsAggregation = resultsAggregation.concat(events);
-    }
-
-    return resultsAggregation;
-  }
-
-  /**
-   *
-   * @param events A list of events to go over and check contract address. Events needing contract change will
-   * have the change applied.
-   */
-  changeContractAddresses(events) {
-    for (let event of events) {
-      for (const contractOverwrite of this.contractsOverwriteArray) {
-        if (this.isContractMatchesExactList(event, contractOverwrite)) {
-          this.editAddressAndAmount(event, contractOverwrite);
-          break;
-        }
+/**
+ *
+ * @param events A list of events to go over and check contract address. Events needing contract change will
+ * have the change applied.
+ */
+function changeContractAddresses(events, contractsOverwriteArray) {
+  for (const event of events) {
+    for (const contractOverwrite of contractsOverwriteArray) {
+      if (contractOverwrite.oldAddresses.includes(event.contract)) {
+        editAddressAndAmount(event, contractOverwrite);
+        break;
       }
     }
   }
+}
 
+function editAddressAndAmount(event, contractOverwrite) {
+  const multiplier = contractOverwrite.mapAddressToMultiplier[event.contract];
+  if (!multiplier) {
+    throw new Error('Event contract does not match expected');
+  }
+  event.contract = contractOverwrite.newContract;
   /**
+   * Note 1: Whether we should round here is up for discussion. The amounts in our pipeline are 'float' anyways but up until this feature
```

Comment:
Shouldn't this be a ticket in itself? Don't think this belongs in the code. Maybe leave a comment that it could be changed or how it works now, but 5 lines of notes is overwhelming for a code file IMO

Reply:
I don't necessarily dislike 5 lines of comments, but they're in the middle of the function. But I do agree, that we need to have more serious docs